crates/router/src/core/payment_link.rs

```diff
@@ -121,9 +123,15 @@ pub async fn initiate_payment_link_flow(
         payment_intent.currency,
         payment_intent.client_secret.clone(),
     )?;
-    let amount = currency
-        .to_currency_base_unit(payment_intent.amount.get_amount_as_i64())
-        .change_context(errors::ApiErrorResponse::CurrencyConversionFailed)?;
+
+    let required_conversion_type = StringMajorUnitForConnector;
```

Comment:
We should have a different type for this like `StringMajorUnit`

Reply:
i cant use pub struct `StringMajorUnit`  since we are already using it for conversion type pub struct `StringMajorUnit(String)`

Reply:
done, added `StringMajorUnitForCore` type to impl on!